grpc/client_test.go

```diff
@@ -0,0 +1,40 @@
+package grpc
+
+import (
+	"crypto/tls"
+	"testing"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/cmd"
+	"google.golang.org/grpc"
+	_ "google.golang.org/grpc/health"
+)
+
+func TestClientSetup(t *testing.T) {
+	tests := []struct {
+		name       string
+		cfg        *cmd.GRPCClientConfig
+		wantTarget string
+		wantErr    bool
+	}{
+		{"valid, address provided", &cmd.GRPCClientConfig{ServerAddress: "localhost:8080"}, "dns:///localhost:8080", false},
+		{"valid, implicit localhost with port provided", &cmd.GRPCClientConfig{ServerAddress: ":8080"}, "dns:///:8080", false},
+		{"valid, two addresses provided", &cmd.GRPCClientConfig{ServerIPAddresses: []string{"127.0.0.1:8080", "127.0.0.2:8080"}}, "static:///127.0.0.1:8080,127.0.0.2:8080", false},
+		{"valid, two addresses provided, one has an implicit localhost, ", &cmd.GRPCClientConfig{ServerIPAddresses: []string{":8080", "127.0.0.2:8080"}}, "static:///:8080,127.0.0.2:8080", false},
+		{"invalid, both address and addresses provided", &cmd.GRPCClientConfig{ServerAddress: "localhost:8080", ServerIPAddresses: []string{"127.0.0.1:8080"}}, "", true},
+		{"invalid, no address or addresses provided", &cmd.GRPCClientConfig{}, "", true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			client, err := ClientSetup(tt.cfg, &tls.Config{}, clientMetrics{}, clock.NewFake(), []grpc.UnaryClientInterceptor{}...)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ClientSetup() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if tt.wantTarget != "" && client.Target() != tt.wantTarget {
```

Comment:
I'd make this just `test.AssertEqual(t, tt.wantTarget, client.Target())`, since `wantTarget` is always specified (and always should be).

I'd also consider renaming it (maybe `expectedTarget`) because it's not a boolean like `wantErr` is, but that's less important.

Reply:
I've gone ahead made the changes. FYI, I use a VSCode built-in generator for making these unit tests so that's why it looks clunky.